File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    category = {"R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0}
    answer = ''
    def give_score(agree, disagree, choice):
        score = choice - 4
        
        if score > 0:
            category[agree] += score
        elif score < 0:
            category[disagree] -= score
            
    def get_result():
        result = ""
        if category["R"] >= category["T"]:
            result += "R"
        else:
            result += "T"
        
        if category["C"] >= category["F"]:
            result += "C"
        else:
            result += "F"
        
        if category["J"] >= category["M"]:
            result += "J"
        else:
            result += "M"
        
        if category["A"] >= category["N"]:
            result += "A"
        else:
            result += "N"
            
        return result
    
    for s, choice in zip(survey, choices):
        give_score(s[1], s[0], choice)

    
    return get_result()

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    const type = [["R", "T"], ["C", "F"], ["J", "M"], ["A", "N"]]
    const result = {"R": 0,"T":0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0};
    survey.forEach((type, idx) => {
        const [disagree, agree] = type.split("");
        const score = choices[idx] - 4;
        if (score < 0) result[disagree] -= score;
        else result[agree] += score;
    })
    
    const answer = type.map(([a, b]) => result[a] >= [result[b]] ? a : b).join("");
    
    return answer
}